shap/explainers/_tree.py

```diff
@@ -991,6 +996,12 @@ def __init__(self, model, data=None, data_missing=None, model_output=None):
                     self.model_output = "probability_doubled" # with predict_proba we need to double the outputs to match
                 else:
                     self.model_output = "probability"
+            # Some properties of the sklearn API are passed to a DMatrix object in xgboost
+            # We need to make sure we do the same here - gh3313
+            properties_to_pass = ["missing", "n_jobs", "enable_categorical", "feature_types"]
```

Comment:
Can we have a function like this at someplace:
```
    def get_dmatrix_properties(model):
        properties_to_pass = ["missing", "n_jobs", "enable_categorical", "feature_types"]
        return {attribute: getattr(model, f'dmatrix_{attribute}') for attribute in properties_to_pass if getattr(model, f'dmatrix_{attribute}') is not None}
```

And then just call this function at all 3 places where you duplicated code and do
`self.dmatrix_props = get_dmatrix_properties(model)`

Reply:
Sure. I'll amend. I was thinking the same but since the TreeEnsemble class was meant to tackle all possible model types I didn't feel it was right creating a class method. But a general method is certainly a better idea :)